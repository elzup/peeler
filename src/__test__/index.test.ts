import m from '../index'

test('works', () => {
  expect(m('before(hit)after')).toMatchSnapshot('single')
  expect(m('aa(bb(cc(dd)cc)bb)aa')).toMatchSnapshot('three nest')
})

test('complex', () => {
  expect(m('')).toMatchSnapshot('empty')
  expect(m('()')).toMatchSnapshot('sub empty')
  expect(m('(())')).toMatchSnapshot('sub sub empty')
  expect(m('({}fuga{})')).toMatchSnapshot('nest side')
  expect(m('aa({cc}bb{c2(dd)c2}b)aa(b2)aa')).toMatchSnapshot('gata gata')
  expect(m('[({})]')).toMatchSnapshot('any bracket')
  expect(
    m(`hel(lo
go)od by`)
  ).toMatchSnapshot('with newline')
})

test('escape works', () => {
  expect(m('\\')).toMatchSnapshot()
  expect(m('(\\(OK\\)\\[GOOD\\])')).toMatchSnapshot()
})

test('bracket options', () => {
  expect(
    m('aaa(bbb{})ccc[ddd]', {
      pairs: ['()'],
    })
  ).toMatchSnapshot('() only')
  expect(
    m('tOshinO kyoko', {
      pairs: ['Oo'],
    })
  ).toMatchSnapshot('rare option')
})

test('nestMax options', () => {
  expect(() =>
    m('((()))', {
      nestMax: 2,
    })
  ).toThrowErrorMatchingInlineSnapshot(
    `"NestError: over nest max limit. options: { nestMax: '2' }"`
  )
  expect(() =>
    m('((()))', {
      nestMax: 1,
    })
  ).toThrowErrorMatchingInlineSnapshot(
    `"NestError: over nest max limit. options: { nestMax: '1' }"`
  )
})

test('bracket parse error', () => {
  expect(() => m('hog{e(b}c)d')).toThrowErrorMatchingInlineSnapshot(
    `"ParseError: 404 pair '{' :7"`
  )
  expect(() => m('(')).toThrowErrorMatchingInlineSnapshot(
    `"ParseError: 404 pair '(' :0"`
  )
})

test('argument error', () => {
  // @ts-ignore: argument test
  expect(() => m(10)).toThrowErrorMatchingInlineSnapshot(
    `"Expected a string, got number"`
  )
  expect(() =>
    m('hoge()', { pairs: ['a'] })
  ).toThrowErrorMatchingInlineSnapshot(
    `"Option error, pairs expected ['[]', '()'...], got [\\"a\\"]"`
  )
})
