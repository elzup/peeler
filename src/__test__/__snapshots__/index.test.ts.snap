// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`argument error 1`] = `"Expected a string, got number"`;

exports[`argument error 2`] = `"Option error, pairs expected ['[]', '()'...], got [\\"a\\"]"`;

exports[`bracket options: () only 1`] = `
Array [
  Object {
    "content": "aaa",
    "nodeType": "text",
    "pos": Object {
      "depth": 0,
      "end": 3,
      "start": 0,
    },
  },
  Object {
    "close": ")",
    "content": "(bbb{})",
    "innerContent": "bbb{}",
    "nodeType": "bracket",
    "nodes": Array [
      Object {
        "content": "bbb{}",
        "nodeType": "text",
        "pos": Object {
          "depth": 1,
          "end": 9,
          "start": 4,
        },
      },
    ],
    "open": "(",
    "pos": Object {
      "depth": 0,
      "end": 9,
      "start": 3,
    },
  },
  Object {
    "content": "ccc[ddd]",
    "nodeType": "text",
    "pos": Object {
      "depth": 0,
      "end": 18,
      "start": 10,
    },
  },
]
`;

exports[`bracket options: rare option 1`] = `
Array [
  Object {
    "content": "t",
    "nodeType": "text",
    "pos": Object {
      "depth": 0,
      "end": 1,
      "start": 0,
    },
  },
  Object {
    "close": "o",
    "content": "OshinO kyoko",
    "innerContent": "shinO kyok",
    "nodeType": "bracket",
    "nodes": Array [
      Object {
        "content": "shin",
        "nodeType": "text",
        "pos": Object {
          "depth": 1,
          "end": 6,
          "start": 2,
        },
      },
      Object {
        "close": "o",
        "content": "O kyo",
        "innerContent": " ky",
        "nodeType": "bracket",
        "nodes": Array [
          Object {
            "content": " ky",
            "nodeType": "text",
            "pos": Object {
              "depth": 2,
              "end": 10,
              "start": 7,
            },
          },
        ],
        "open": "O",
        "pos": Object {
          "depth": 1,
          "end": 10,
          "start": 6,
        },
      },
      Object {
        "content": "k",
        "nodeType": "text",
        "pos": Object {
          "depth": 1,
          "end": 12,
          "start": 11,
        },
      },
    ],
    "open": "O",
    "pos": Object {
      "depth": 0,
      "end": 12,
      "start": 1,
    },
  },
]
`;

exports[`bracket parse error: entwined 1`] = `"ParseError: 404 pair '{' :7"`;

exports[`bracket parse error: no finish 1`] = `"ParseError: 404 pair '(' :0"`;

exports[`complex: any bracket 1`] = `
Array [
  Object {
    "close": "]",
    "content": "[({})]",
    "innerContent": "({})",
    "nodeType": "bracket",
    "nodes": Array [
      Object {
        "close": ")",
        "content": "({})",
        "innerContent": "{}",
        "nodeType": "bracket",
        "nodes": Array [
          Object {
            "close": "}",
            "content": "{}",
            "innerContent": "",
            "nodeType": "bracket",
            "nodes": Array [],
            "open": "{",
            "pos": Object {
              "depth": 2,
              "end": 3,
              "start": 2,
            },
          },
        ],
        "open": "(",
        "pos": Object {
          "depth": 1,
          "end": 4,
          "start": 1,
        },
      },
    ],
    "open": "[",
    "pos": Object {
      "depth": 0,
      "end": 5,
      "start": 0,
    },
  },
]
`;

exports[`complex: empty 1`] = `Array []`;

exports[`complex: gata gata 1`] = `
Array [
  Object {
    "content": "aa",
    "nodeType": "text",
    "pos": Object {
      "depth": 0,
      "end": 2,
      "start": 0,
    },
  },
  Object {
    "close": ")",
    "content": "({cc}bb{c2(dd)c2}b)",
    "innerContent": "{cc}bb{c2(dd)c2}b",
    "nodeType": "bracket",
    "nodes": Array [
      Object {
        "close": "}",
        "content": "{cc}",
        "innerContent": "cc",
        "nodeType": "bracket",
        "nodes": Array [
          Object {
            "content": "cc",
            "nodeType": "text",
            "pos": Object {
              "depth": 2,
              "end": 6,
              "start": 4,
            },
          },
        ],
        "open": "{",
        "pos": Object {
          "depth": 1,
          "end": 6,
          "start": 3,
        },
      },
      Object {
        "content": "bb",
        "nodeType": "text",
        "pos": Object {
          "depth": 1,
          "end": 9,
          "start": 7,
        },
      },
      Object {
        "close": "}",
        "content": "{c2(dd)c2}",
        "innerContent": "c2(dd)c2",
        "nodeType": "bracket",
        "nodes": Array [
          Object {
            "content": "c2",
            "nodeType": "text",
            "pos": Object {
              "depth": 2,
              "end": 12,
              "start": 10,
            },
          },
          Object {
            "close": ")",
            "content": "(dd)",
            "innerContent": "dd",
            "nodeType": "bracket",
            "nodes": Array [
              Object {
                "content": "dd",
                "nodeType": "text",
                "pos": Object {
                  "depth": 3,
                  "end": 15,
                  "start": 13,
                },
              },
            ],
            "open": "(",
            "pos": Object {
              "depth": 2,
              "end": 15,
              "start": 12,
            },
          },
          Object {
            "content": "c2",
            "nodeType": "text",
            "pos": Object {
              "depth": 2,
              "end": 18,
              "start": 16,
            },
          },
        ],
        "open": "{",
        "pos": Object {
          "depth": 1,
          "end": 18,
          "start": 9,
        },
      },
      Object {
        "content": "b",
        "nodeType": "text",
        "pos": Object {
          "depth": 1,
          "end": 20,
          "start": 19,
        },
      },
    ],
    "open": "(",
    "pos": Object {
      "depth": 0,
      "end": 20,
      "start": 2,
    },
  },
  Object {
    "content": "aa",
    "nodeType": "text",
    "pos": Object {
      "depth": 0,
      "end": 23,
      "start": 21,
    },
  },
  Object {
    "close": ")",
    "content": "(b2)",
    "innerContent": "b2",
    "nodeType": "bracket",
    "nodes": Array [
      Object {
        "content": "b2",
        "nodeType": "text",
        "pos": Object {
          "depth": 1,
          "end": 26,
          "start": 24,
        },
      },
    ],
    "open": "(",
    "pos": Object {
      "depth": 0,
      "end": 26,
      "start": 23,
    },
  },
  Object {
    "content": "aa",
    "nodeType": "text",
    "pos": Object {
      "depth": 0,
      "end": 29,
      "start": 27,
    },
  },
]
`;

exports[`complex: nest side 1`] = `
Array [
  Object {
    "close": ")",
    "content": "({}fuga{})",
    "innerContent": "{}fuga{}",
    "nodeType": "bracket",
    "nodes": Array [
      Object {
        "close": "}",
        "content": "{}",
        "innerContent": "",
        "nodeType": "bracket",
        "nodes": Array [],
        "open": "{",
        "pos": Object {
          "depth": 1,
          "end": 2,
          "start": 1,
        },
      },
      Object {
        "content": "fuga",
        "nodeType": "text",
        "pos": Object {
          "depth": 1,
          "end": 7,
          "start": 3,
        },
      },
      Object {
        "close": "}",
        "content": "{}",
        "innerContent": "",
        "nodeType": "bracket",
        "nodes": Array [],
        "open": "{",
        "pos": Object {
          "depth": 1,
          "end": 8,
          "start": 7,
        },
      },
    ],
    "open": "(",
    "pos": Object {
      "depth": 0,
      "end": 9,
      "start": 0,
    },
  },
]
`;

exports[`complex: sub empty 1`] = `
Array [
  Object {
    "close": ")",
    "content": "()",
    "innerContent": "",
    "nodeType": "bracket",
    "nodes": Array [],
    "open": "(",
    "pos": Object {
      "depth": 0,
      "end": 1,
      "start": 0,
    },
  },
]
`;

exports[`complex: sub sub empty 1`] = `
Array [
  Object {
    "close": ")",
    "content": "(())",
    "innerContent": "()",
    "nodeType": "bracket",
    "nodes": Array [
      Object {
        "close": ")",
        "content": "()",
        "innerContent": "",
        "nodeType": "bracket",
        "nodes": Array [],
        "open": "(",
        "pos": Object {
          "depth": 1,
          "end": 2,
          "start": 1,
        },
      },
    ],
    "open": "(",
    "pos": Object {
      "depth": 0,
      "end": 3,
      "start": 0,
    },
  },
]
`;

exports[`complex: with newline 1`] = `
Array [
  Object {
    "content": "hel",
    "nodeType": "text",
    "pos": Object {
      "depth": 0,
      "end": 3,
      "start": 0,
    },
  },
  Object {
    "close": ")",
    "content": "(lo
go)",
    "innerContent": "lo
go",
    "nodeType": "bracket",
    "nodes": Array [
      Object {
        "content": "lo
go",
        "nodeType": "text",
        "pos": Object {
          "depth": 1,
          "end": 9,
          "start": 4,
        },
      },
    ],
    "open": "(",
    "pos": Object {
      "depth": 0,
      "end": 9,
      "start": 3,
    },
  },
  Object {
    "content": "od by",
    "nodeType": "text",
    "pos": Object {
      "depth": 0,
      "end": 15,
      "start": 10,
    },
  },
]
`;

exports[`escape works 1`] = `
Array [
  Object {
    "content": "\\\\",
    "nodeType": "text",
    "pos": Object {
      "depth": 0,
      "end": 1,
      "start": 0,
    },
  },
]
`;

exports[`escape works 2`] = `
Array [
  Object {
    "close": ")",
    "content": "(\\\\(OK\\\\)\\\\[GOOD\\\\])",
    "innerContent": "\\\\(OK\\\\)\\\\[GOOD\\\\]",
    "nodeType": "bracket",
    "nodes": Array [
      Object {
        "content": "(OK)[GOOD]",
        "nodeType": "text",
        "pos": Object {
          "depth": 1,
          "end": 15,
          "start": 1,
        },
      },
    ],
    "open": "(",
    "pos": Object {
      "depth": 0,
      "end": 15,
      "start": 0,
    },
  },
]
`;

exports[`nestMax options: stop nest 1 1`] = `"NestError: over nest max limit. options: { nestMax: '1' }"`;

exports[`nestMax options: stop nest 2 1`] = `"NestError: over nest max limit. options: { nestMax: '2' }"`;

exports[`works: single 1`] = `
Array [
  Object {
    "content": "before",
    "nodeType": "text",
    "pos": Object {
      "depth": 0,
      "end": 6,
      "start": 0,
    },
  },
  Object {
    "close": ")",
    "content": "(hit)",
    "innerContent": "hit",
    "nodeType": "bracket",
    "nodes": Array [
      Object {
        "content": "hit",
        "nodeType": "text",
        "pos": Object {
          "depth": 1,
          "end": 10,
          "start": 7,
        },
      },
    ],
    "open": "(",
    "pos": Object {
      "depth": 0,
      "end": 10,
      "start": 6,
    },
  },
  Object {
    "content": "after",
    "nodeType": "text",
    "pos": Object {
      "depth": 0,
      "end": 16,
      "start": 11,
    },
  },
]
`;

exports[`works: three nest 1`] = `
Array [
  Object {
    "content": "aa",
    "nodeType": "text",
    "pos": Object {
      "depth": 0,
      "end": 2,
      "start": 0,
    },
  },
  Object {
    "close": ")",
    "content": "(bb(cc(dd)cc)bb)",
    "innerContent": "bb(cc(dd)cc)bb",
    "nodeType": "bracket",
    "nodes": Array [
      Object {
        "content": "bb",
        "nodeType": "text",
        "pos": Object {
          "depth": 1,
          "end": 5,
          "start": 3,
        },
      },
      Object {
        "close": ")",
        "content": "(cc(dd)cc)",
        "innerContent": "cc(dd)cc",
        "nodeType": "bracket",
        "nodes": Array [
          Object {
            "content": "cc",
            "nodeType": "text",
            "pos": Object {
              "depth": 2,
              "end": 8,
              "start": 6,
            },
          },
          Object {
            "close": ")",
            "content": "(dd)",
            "innerContent": "dd",
            "nodeType": "bracket",
            "nodes": Array [
              Object {
                "content": "dd",
                "nodeType": "text",
                "pos": Object {
                  "depth": 3,
                  "end": 11,
                  "start": 9,
                },
              },
            ],
            "open": "(",
            "pos": Object {
              "depth": 2,
              "end": 11,
              "start": 8,
            },
          },
          Object {
            "content": "cc",
            "nodeType": "text",
            "pos": Object {
              "depth": 2,
              "end": 14,
              "start": 12,
            },
          },
        ],
        "open": "(",
        "pos": Object {
          "depth": 1,
          "end": 14,
          "start": 5,
        },
      },
      Object {
        "content": "bb",
        "nodeType": "text",
        "pos": Object {
          "depth": 1,
          "end": 17,
          "start": 15,
        },
      },
    ],
    "open": "(",
    "pos": Object {
      "depth": 0,
      "end": 17,
      "start": 2,
    },
  },
  Object {
    "content": "aa",
    "nodeType": "text",
    "pos": Object {
      "depth": 0,
      "end": 20,
      "start": 18,
    },
  },
]
`;
